### ISSUE TITLE
Segmentation fault with a corrupted GIF image

### ISSUE THREAD

**Issue by kleisauke** [ImageMagick/ImageMagick#926]
When loading a corrupted GIF image through ImageMagick 6 it segfaults. 

Test image: https://t0.nl/pixel.gif

Reproducible with:
```bash
magick identify pixel.gif
```

GDB backtrace:
```
#0  0x00007ffff4a5f4c5 in fseeko (fp=0x632eb0, offset=offset@entry=0, whence=whence@entry=1) at fseeko.c:38
#1  0x00007ffff7967820 in IsBlobSeekable (image=image@entry=0x62d610) at magick/blob.c:2135
#2  0x00007ffff7a3acb9 in SetImageInfo (image_info=image_info@entry=0x621af0, frames=frames@entry=0, exception=exception@entry=0x609190) at magick/image.c:2912
#3  0x00007ffff799efc3 in ReadImage (image_info=image_info@entry=0x61b430, exception=exception@entry=0x609190) at magick/constitute.c:398
#4  0x00007ffff7ace2a9 in ReadStream (image_info=image_info@entry=0x617290, stream=stream@entry=0x7ffff799e920 <PingStream>, exception=exception@entry=0x609190) at magick/stream.c:1009
#5  0x00007ffff799ec11 in PingImage (image_info=image_info@entry=0x612f50, exception=exception@entry=0x609190) at magick/constitute.c:226
#6  0x00007ffff799ee4b in PingImages (image_info=image_info@entry=0x612f50, exception=exception@entry=0x609190) at magick/constitute.c:321
#7  0x00007ffff767074c in IdentifyImageCommand (image_info=0x60edb0, argc=2, argv=0x60a630, metadata=0x7fffffffe500, exception=0x609190) at wand/identify.c:324
#8  0x00007ffff769f203 in MagickCommandGenesis (image_info=image_info@entry=0x60a930, command=0x400960 <IdentifyImageCommand@plt>, argc=argc@entry=2, argv=argv@entry=0x7fffffffe618, metadata=metadata@entry=0x7fffffffe500,
    exception=exception@entry=0x609190) at wand/mogrify.c:172
#9  0x0000000000400a2f in IdentifyMain (argv=0x7fffffffe618, argc=2) at utilities/identify.c:80
#10 main (argc=2, argv=0x7fffffffe618) at utilities/identify.c:93
```

GDB backtrace (full):
<details>
 <summary>Click for details</summary>

```
#0  0x00007ffff4a5f4c5 in fseeko (fp=0x632eb0, offset=offset@entry=0, whence=whence@entry=1) at fseeko.c:38
        __self = 0x7ffff7fd6800
        _IO_acquire_lock_file = 0x632eb0
        result = <optimized out>
#1  0x00007ffff7967820 in IsBlobSeekable (image=image@entry=0x62d610) at magick/blob.c:2135
        status = <optimized out>
        blob_info = <optimized out>
        __PRETTY_FUNCTION__ = "IsBlobSeekable"
        __func__ = "IsBlobSeekable"
#2  0x00007ffff7a3acb9 in SetImageInfo (image_info=image_info@entry=0x621af0, frames=frames@entry=0, exception=exception@entry=0x609190) at magick/image.c:2912
        extension = "gif\000l.gif", '\000' <repeats 4086 times>
        filename = "pixel.gif", '\000' <repeats 4086 times>
        magic = "GIF\000l.gif", '\000' <repeats 4086 times>
        q = 0x0
        subimage = "\000ixel.gif", '\000' <repeats 4086 times>
        magic_info = <optimized out>
        magick_info = <optimized out>
        sans_exception = <optimized out>
        image = 0x62d610
        status = <optimized out>
        p = <optimized out>
        count = <optimized out>
        magick = '\000' <repeats 344 times>...
        __PRETTY_FUNCTION__ = "SetImageInfo"
        __func__ = "SetImageInfo"
#3  0x00007ffff799efc3 in ReadImage (image_info=image_info@entry=0x61b430, exception=exception@entry=0x609190) at magick/constitute.c:398
        filename = '\000' <repeats 16 times>, "F\000\000\000\000\000\000\000\000\020\000\000\000\000\000\000\b\000\000\000\000\000\000\000\304\023JZ\000\000\000\000S`d9\000\000\000\000\302WIZ\000\000\000\000j\246_\017\000\000\000\000\270\023JZ\000\000\000\000\023\372\364\030", '\000' <repeats 28 times>, "\060\246`\000\000\000\000\000\000p\207\336\312\300U\003\260\212\377\377\377\177\000\000iJ\255\367\377\177\000\000\060\246`\000\000\000\000\000\016I\256\367\377\177", '\000' <repeats 26 times>, "\002\000\000\000\000\000\000\000l\333\377\377\377\177\000\000"...
        magick = '\000' <repeats 208 times>...
        magick_filename = "pixel.gif", '\000' <repeats 271 times>...
        value = <optimized out>
        delegate_info = <optimized out>
        magick_info = <optimized out>
        sans_exception = <optimized out>
        geometry_info = {rho = 0, sigma = 0, xi = 2.5251749917921463e-312, psi = 6.9533489118153346e-310, chi = 6.9533558061337881e-310}
        image = <optimized out>
        next = 0x355c0cade877000
        read_info = 0x621af0
        flags = <optimized out>
        thread_support = <optimized out>
        __PRETTY_FUNCTION__ = "ReadImage"
        __func__ = "ReadImage"
#4  0x00007ffff7ace2a9 in ReadStream (image_info=image_info@entry=0x617290, stream=stream@entry=0x7ffff799e920 <PingStream>, exception=exception@entry=0x609190) at magick/stream.c:1009
        cache_methods = {get_virtual_pixel_handler = 0x7ffff7acda90 <GetVirtualPixelStream>, get_virtual_pixels_handler = 0x7ffff7acd280 <GetVirtualPixelsStream>,
          get_virtual_indexes_from_handler = 0x7ffff7acd1a0 <GetVirtualIndexesFromStream>, get_one_virtual_pixel_from_handler = 0x7ffff7acde20 <GetOneVirtualPixelFromStream>,
          get_authentic_pixels_handler = 0x7ffff7acd920 <GetAuthenticPixelsStream>, get_authentic_indexes_from_handler = 0x7ffff7accfe0 <GetAuthenticIndexesFromStream>,
          get_one_authentic_pixel_from_handler = 0x7ffff7acda00 <GetOneAuthenticPixelFromStream>, get_authentic_pixels_from_handler = 0x7ffff7acd0c0 <GetAuthenticPixelsFromStream>,
          queue_authentic_pixels_handler = 0x7ffff7acd520 <QueueAuthenticPixelsStream>, sync_authentic_pixels_handler = 0x7ffff7ac9fc0 <SyncAuthenticPixelsStream>, destroy_pixel_handler = 0x7ffff7acd360 <DestroyPixelStream>}
        image = <optimized out>
        read_info = 0x61b430
        __PRETTY_FUNCTION__ = "ReadStream"
        __func__ = "ReadStream"
#5  0x00007ffff799ec11 in PingImage (image_info=image_info@entry=0x612f50, exception=exception@entry=0x609190) at magick/constitute.c:226
        image = <optimized out>
        ping_info = 0x617290
        __PRETTY_FUNCTION__ = "PingImage"
        __func__ = "PingImage"
#6  0x00007ffff799ee4b in PingImages (image_info=image_info@entry=0x612f50, exception=exception@entry=0x609190) at magick/constitute.c:321
        filename = "pixel.gif\000\377\377\377\177\000\000*\313\377\377\377\177\000\000\065N\337\367\377\177\000\000\327-\200\363\377\177\000\000\250\246\375\367\377\177\000\000\001\000\000\000\000\000\000\000 \324\377\377\377\177\000\000 \000\000\000\000\000\000\000\243\206\336\367\377\177\000\000\000\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\375\367\377\177\000\000\345\216\336\367\377\177\000\000\001\000\000\000\000\000\000\000P\240\375\367\377\177\000\000\000\000\000\000\000\000\000\000
hO\322\363\377\177\000\000(\275\257\363\377\177\000\000"...
        image = <optimized out>
        images = 0xffffffff94400000
        read_info = <optimized out>
        __PRETTY_FUNCTION__ = "PingImages"
        __func__ = "PingImages"
#7  0x00007ffff767074c in IdentifyImageCommand (image_info=0x60edb0, argc=2, argv=0x60a630, metadata=0x7fffffffe500, exception=0x609190) at wand/identify.c:324
        filename = 0x60a330 "pixel.gif"
        images = <optimized out>
        identify_info = 0x612f50
        format = 0x0
        option = <optimized out>
        image = 0x0
        image_stack = {{image_info = 0x60edb0, image = 0x0}, {image_info = 0x0, image = 0x0} <repeats 29 times>, {image_info = 0x0, image = 0x7ffff7de45c4 <check_match+292>}, {image_info = 0x0, image = 0x7ffff7ffa1a8}, {
            image_info = 0x0, image = 0x1}, {image_info = 0xb01bca00, image = 0x7ffff7de4cfb <do_lookup_x+1803>}, {image_info = 0x0, image = 0x7fffffffdde0}, {image_info = 0x7ffff7ffa188, image = 0x7ffff7ffa1d8}, {
            image_info = 0x7fffffffdef0, image = 0x7fffffffdee0}, {image_info = 0x0, image = 0x0}, {image_info = 0x0, image = 0x7ffff7de44f9 <check_match+89>}, {image_info = 0x7ffff7ffe6e0, image = 0x7ffff75ea788}, {image_info = 0x2,
            image = 0x21}, {image_info = 0x24ce7d03, image = 0x7ffff7de4cfb <do_lookup_x+1803>}, {image_info = 0x0, image = 0x7fffffffde60}, {image_info = 0x7ffff75e9da0, image = 0x7ffff75ee478}, {image_info = 0x7fffffffdf70,
            image = 0x7fffffffdf60}, {image_info = 0x3, image = 0x7ffff7de4cfb <do_lookup_x+1803>}, {image_info = 0x0, image = 0x0}, {image_info = 0x7ffff7ff7b48, image = 0x4005e7}, {image_info = 0x7ffff75f1fe8, image = 0x400470}, {
            image_info = 0x100000000, image = 0x10000028a}, {image_info = 0x0, image = 0x7fffffffe028}, {image_info = 0x7fffffffe000, image = 0x1}, {image_info = 0x0, image = 0x7ffff7ffe4a8}, {image_info = 0x7ffff7ffe150,
            image = 0x7ffff7de4fdf <_dl_lookup_symbol_x+287>}, {image_info = 0x0, image = 0x0}, {image_info = 0x7fff00000001, image = 0x0}, {image_info = 0x1, image = 0x7ffff7ffe150}, {image_info = 0x180, image = 0x7fff0000000d}, {
            image_info = 0x3, image = 0x0}, {image_info = 0x0, image = 0x7ffff7ffe4a8}, {image_info = 0x7fffffffdf70, image = 0x7fffffffdf60}, {image_info = 0x24ce7d03, image = 0x4005e7}, {image_info = 0xffffffff, image = 0x0}, {
            image_info = 0x7ffff75ee478, image = 0x7ffff7ff7b48}, {image_info = 0x7fff0000007c, image = 0x0}, {image_info = 0x7fffffffdfbf, image = 0x1000002af}, {image_info = 0x7c, image = 0x7fffffffe128}, {
            image_info = 0x7fff0000007c, image = 0x7ffff4da7760 <main_arena>}, {image_info = 0xc0, image = 0xc}, {image_info = 0xc, image = 0x602040 <IdentifyImageCommand@got.plt>}, {image_info = 0x1, image = 0x1}, {image_info = 0x0,
            image = 0x7fffffffe618}, {image_info = 0x7fffffffe500, image = 0x7ffff7de98de <_dl_fixup+206>}, {image_info = 0x1, image = 0x0}, {image_info = 0x7fffffffe1a0, image = 0x7ffff75ee478}, {image_info = 0x7fffffffe3c0,
            image = 0x7ffff7df0f5a <_dl_runtime_resolve_xsave+138>}, {image_info = 0x400960 <IdentifyImageCommand@plt>, image = 0x7fffffffe500}, {image_info = 0x7fffffffe618, image = 0x2}, {image_info = 0x60a930, image = 0x609190}, {
            image_info = 0x60a570, image = 0xf7ff7b48}, {image_info = 0x7fffffffe2c0, image = 0x7fffecc2f000}, {image_info = 0x666, image = 0xffff00001fa0}, {image_info = 0x7fffecc552e0, image = 0x3e964}, {image_info = 0x7fffecdc36d0,
            image = 0x36}, {image_info = 0x7fffecdc3710, image = 0x996}, {image_info = 0x7fffecc93c50, image = 0x12fa76}, {image_info = 0x7fffecdc40b0, image = 0x11e}, {image_info = 0x7fffecdc41d0, image = 0x39}, {
            image_info = 0x60a930, image = 0x7ffff7de98de <_dl_fixup+206>}, {image_info = 0x7fffecdc4210, image = 0x22}, {image_info = 0x0, image = 0xffff0000ff000000}, {image_info = 0x54552e53555f6e65, image = 0x7fff00382d46}, {
            image_info = 0x2e2e00544d470043, image = 0x5249445a54002f}, {image_info = 0x0, image = 0x0}, {image_info = 0x4040404040404040, image = 0x4040404040404040}, {image_info = 0x5a5a5a5a5a5a5a5a, image = 0x5a5a5a5a5a5a5a5a}, {
            image_info = 0x2020202020202020, image = 0x2020202020202020}, {image_info = 0xffffff0000, image = 0x0}, {image_info = 0x0, image = 0x0}, {image_info = 0xffff00ffffff0000, image = 0xff00ffffffffffff}, {image_info = 0x0,
            image = 0x0}, {image_info = 0xffff000000ff00ff, image = 0xff000000ffff00}, {image_info = 0x0, image = 0x0}, {image_info = 0x0, image = 0x0}, {image_info = 0x0, image = 0x0}, {image_info = 0x0, image = 0x0}, {
            image_info = 0x10f, image = 0x7ffff4a16ae0 <new_composite_name+416>}, {image_info = 0x7ffff4b52c40 <_nl_category_names>, image = 0x100608fd0}, {image_info = 0x7fffffffe330, image = 0x7ffff4a23102 <__GI_getenv+194>}, {
            image_info = 0x0, image = 0x1}, {image_info = 0x7ffff4b69fd5 <_nl_C_name>, image = 0x60ec90}, {image_info = 0x7ffff4da8060 <_nl_global_locale>, image = 0x7ffff4b69fd5 <_nl_C_name>}, {image_info = 0x2, image = 0x0}, {
            image_info = 0x0, image = 0x0} <repeats 15 times>, {image_info = 0x60a930, image = 0x0}}
        fire = <optimized out>
        pend = MagickFalse
        respect_parenthesis = MagickFalse
        status = 1
        i = 1
        count = <optimized out>
        j = 1
        k = 0
        __PRETTY_FUNCTION__ = "IdentifyImageCommand"
        __func__ = "IdentifyImageCommand"
#8  0x00007ffff769f203 in MagickCommandGenesis (image_info=image_info@entry=0x60a930, command=0x400960 <IdentifyImageCommand@plt>, argc=argc@entry=2, argv=argv@entry=0x7fffffffe618, metadata=metadata@entry=0x7fffffffe500,
    exception=exception@entry=0x609190) at wand/mogrify.c:172
        option = <optimized out>
        duration = <optimized out>
        serial = <optimized out>
        concurrent = <optimized out>
        regard_warnings = <optimized out>
        status = MagickTrue
        i = <optimized out>
        iterations = <optimized out>
        number_threads = <optimized out>
        n = <optimized out>
#9  0x0000000000400a2f in IdentifyMain (argv=0x7fffffffe618, argc=2) at utilities/identify.c:80
        metadata = 0x0
        exception = 0x609190
        image_info = 0x60a930
        status = <optimized out>
#10 main (argc=2, argv=0x7fffffffe618) at utilities/identify.c:93
```
</details>
&nbsp;

Tested with ImageMagick 6.9.9 patch level 30 and giflib 4.1.6.

**Comment by urban-warrior** [comment]
Thanks for the problem report.  We can reproduce it and will have a patch to fix it in GIT master branch @ https://github.com/ImageMagick/ImageMagick later today.  The patch will be available in the beta releases of ImageMagick @ https://www.imagemagick.org/download/beta/ by sometime tomorrow.

**Comment by kleisauke** [comment]
I confirm that this segfault has been resolved in ImageMagick 6.9.9 patch level 31. Thanks for fixing this! I'll close this issue.

**Comment by kleisauke** [comment]
The same segmentation fault occurs with ImageMagick 6.9.9 patch level 34. It looks like https://github.com/ImageMagick/ImageMagick/commit/9d0892bf0e87f893c9a52eb703cbcded7cd2af6f has re-introduced this segfault (issue https://github.com/ImageMagick/ImageMagick/issues/945).

GDB backtrace:
```
#0  0x00007ffff4a5e4c5 in fseeko (fp=0x632eb0, offset=offset@entry=0, whence=whence@entry=1) at fseeko.c:38
#1  0x00007ffff796680f in IsBlobSeekable (image=image@entry=0x62d610) at magick/blob.c:2137
#2  0x00007ffff7a39f89 in SetImageInfo (image_info=image_info@entry=0x621af0, frames=frames@entry=0, exception=exception@entry=0x609190) at magick/image.c:2912
#3  0x00007ffff799dfd3 in ReadImage (image_info=image_info@entry=0x61b430, exception=exception@entry=0x609190) at magick/constitute.c:398
#4  0x00007ffff7acd5e9 in ReadStream (image_info=image_info@entry=0x617290, stream=stream@entry=0x7ffff799d930 <PingStream>, exception=exception@entry=0x609190) at magick/stream.c:1009
#5  0x00007ffff799dc21 in PingImage (image_info=image_info@entry=0x612f50, exception=exception@entry=0x609190) at magick/constitute.c:226
#6  0x00007ffff799de5b in PingImages (image_info=image_info@entry=0x612f50, exception=exception@entry=0x609190) at magick/constitute.c:321
#7  0x00007ffff766f74c in IdentifyImageCommand (image_info=0x60edb0, argc=2, argv=0x60a630, metadata=0x7fffffffe4f0, exception=0x609190) at wand/identify.c:324
#8  0x00007ffff769e203 in MagickCommandGenesis (image_info=image_info@entry=0x60a930, command=0x400960 <IdentifyImageCommand@plt>, argc=argc@entry=2, argv=argv@entry=0x7fffffffe608, metadata=metadata@entry=0x7fffffffe4f0,
    exception=exception@entry=0x609190) at wand/mogrify.c:172
#9  0x0000000000400a2f in IdentifyMain (argv=0x7fffffffe608, argc=2) at utilities/identify.c:80
#10 main (argc=2, argv=0x7fffffffe608) at utilities/identify.c:93
```

GDB backtrace (full):
<details>
 <summary>Click for details</summary>

```
#0  0x00007ffff4a5e4c5 in fseeko (fp=0x632eb0, offset=offset@entry=0, whence=whence@entry=1) at fseeko.c:38
        __self = 0x7ffff7fd5800
        _IO_acquire_lock_file = 0x632eb0
        result = <optimized out>
#1  0x00007ffff796680f in IsBlobSeekable (image=image@entry=0x62d610) at magick/blob.c:2137
        status = <optimized out>
        blob_info = <optimized out>
        __PRETTY_FUNCTION__ = "IsBlobSeekable"
        __func__ = "IsBlobSeekable"
#2  0x00007ffff7a39f89 in SetImageInfo (image_info=image_info@entry=0x621af0, frames=frames@entry=0, exception=exception@entry=0x609190) at magick/image.c:2912
        extension = "gif\000l.gif", '\000' <repeats 4086 times>
        filename = "pixel.gif", '\000' <repeats 4086 times>
        magic = "GIF\000l.gif", '\000' <repeats 4086 times>
        q = 0x0
        subimage = "\000ixel.gif", '\000' <repeats 4086 times>
        magic_info = <optimized out>
        magick_info = <optimized out>
        sans_exception = <optimized out>
        image = 0x62d610
        status = <optimized out>
        p = <optimized out>
        count = <optimized out>
        magick = '\000' <repeats 344 times>...
        __PRETTY_FUNCTION__ = "SetImageInfo"
        __func__ = "SetImageInfo"
#3  0x00007ffff799dfd3 in ReadImage (image_info=image_info@entry=0x61b430, exception=exception@entry=0x609190) at magick/constitute.c:398
        filename = '\000' <repeats 16 times>, "F\000\000\000\000\000\000\000\000\020\000\000\000\000\000\000\b\000\000\000\000\000\000\000\270\277mZ\000\000\000\000V\312M \000\000\000\000\302WIZ", '\000' <repeats 12 times>, "\324$cZ\000\000\000\000O\236Y,", '\000' <repeats 28 times>, "\060\246`\000\000\000\000\000\000B\263I\026m\247\261\240\212\377\377\377\177\000\000\251=\255\367\377\177\000\000\060\246`\000\000\000\000\000N<\256\367\377\177", '\000' <repeats 26 times>, "\002\000\000\000\000\000\000\000\\\333\377\377\377\177\000\000"...
        magick = '\000' <repeats 192 times>, "`\220\377\377\377\177\000\000"...
        magick_filename = "pixel.gif", '\000' <repeats 271 times>...
        value = <optimized out>
        delegate_info = <optimized out>
        magick_info = <optimized out>
        sans_exception = <optimized out>
        geometry_info = {rho = 0, sigma = 0, xi = 2.5251749917921463e-312, psi = 6.9533489116540716e-310, chi = 6.9533558061329976e-310}
        image = <optimized out>
        next = 0xb1a76d1649b34200
        read_info = 0x621af0
        flags = <optimized out>
        thread_support = <optimized out>
        __PRETTY_FUNCTION__ = "ReadImage"
        __func__ = "ReadImage"
#4  0x00007ffff7acd5e9 in ReadStream (image_info=image_info@entry=0x617290, stream=stream@entry=0x7ffff799d930 <PingStream>, exception=exception@entry=0x609190) at magick/stream.c:1009
        cache_methods = {get_virtual_pixel_handler = 0x7ffff7accdd0 <GetVirtualPixelStream>, get_virtual_pixels_handler = 0x7ffff7acc5c0 <GetVirtualPixelsStream>,
          get_virtual_indexes_from_handler = 0x7ffff7acc4e0 <GetVirtualIndexesFromStream>, get_one_virtual_pixel_from_handler = 0x7ffff7acd160 <GetOneVirtualPixelFromStream>,
          get_authentic_pixels_handler = 0x7ffff7accc60 <GetAuthenticPixelsStream>, get_authentic_indexes_from_handler = 0x7ffff7acc320 <GetAuthenticIndexesFromStream>,
          get_one_authentic_pixel_from_handler = 0x7ffff7accd40 <GetOneAuthenticPixelFromStream>, get_authentic_pixels_from_handler = 0x7ffff7acc400 <GetAuthenticPixelsFromStream>,
          queue_authentic_pixels_handler = 0x7ffff7acc860 <QueueAuthenticPixelsStream>, sync_authentic_pixels_handler = 0x7ffff7ac9300 <SyncAuthenticPixelsStream>, destroy_pixel_handler = 0x7ffff7acc6a0 <DestroyPixelStream>}
        image = <optimized out>
        read_info = 0x61b430
        __PRETTY_FUNCTION__ = "ReadStream"
        __func__ = "ReadStream"
#5  0x00007ffff799dc21 in PingImage (image_info=image_info@entry=0x612f50, exception=exception@entry=0x609190) at magick/constitute.c:226
        image = <optimized out>
        ping_info = 0x617290
        __PRETTY_FUNCTION__ = "PingImage"
        __func__ = "PingImage"
#6  0x00007ffff799de5b in PingImages (image_info=image_info@entry=0x612f50, exception=exception@entry=0x609190) at magick/constitute.c:321
        filename = "pixel.gif\000\377\377\377\177\000\000\032\313\377\377\377\177\000\000\065N\337\367\377\177\000\000\327\035\200\363\377\177\000\000\250\226\375\367\377\177\000\000\001\000\000\000\000\000\000\000\020\324\377\377\377\177\000\000 \000\000\000\000\000\000\000\243\206\336\367\377\177\000\000\000\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\001\000\375\367\377\177\000\000\345\216\336\367\377\177\000\000\001\000\000\000\000\000\000\000P\220\375\367\377\177\000\000\000\000\000\000\000\00
0\000\000h?\322\363\377\177\000\000(\255\257\363\377\177\000\000"...
        image = <optimized out>
        images = 0xffffffff94200000
        read_info = <optimized out>
        __PRETTY_FUNCTION__ = "PingImages"
        __func__ = "PingImages"
#7  0x00007ffff766f74c in IdentifyImageCommand (image_info=0x60edb0, argc=2, argv=0x60a630, metadata=0x7fffffffe4f0, exception=0x609190) at wand/identify.c:324
        filename = 0x60a330 "pixel.gif"
        images = <optimized out>
        identify_info = 0x612f50
        format = 0x0
        option = <optimized out>
        image = 0x0
        image_stack = {{image_info = 0x60edb0, image = 0x0}, {image_info = 0x0, image = 0x0} <repeats 30 times>, {image_info = 0x0, image = 0x7ffff7de45c4 <check_match+292>}, {image_info = 0x0, image = 0x7ffff7ffa1a8}, {
            image_info = 0x0, image = 0x1}, {image_info = 0xb01bca00, image = 0x7ffff7de4cfb <do_lookup_x+1803>}, {image_info = 0x0, image = 0x7ffff7de44f9 <check_match+89>}, {image_info = 0x7ffff7ffa188, image = 0x7ffff75e9788}, {
            image_info = 0x2, image = 0x21}, {image_info = 0x24ce7d03, image = 0x7ffff7de4cfb <do_lookup_x+1803>}, {image_info = 0x0, image = 0x7fffffffde20}, {image_info = 0x7ffff75e8da0, image = 0x7ffff75ed478}, {
            image_info = 0x7fffffffdf30, image = 0x7fffffffdf20}, {image_info = 0x3, image = 0x7ffff7de4cfb <do_lookup_x+1803>}, {image_info = 0x0, image = 0x0}, {image_info = 0x7ffff7ff6b48, image = 0x4005e7}, {
            image_info = 0x7ffff75f0fe8, image = 0x400470}, {image_info = 0x100000000, image = 0x10000028a}, {image_info = 0x0, image = 0x7fffffffdfe8}, {image_info = 0x7fffffffdfc0, image = 0x1}, {image_info = 0x0,
            image = 0x7ffff7ffe4a8}, {image_info = 0x7ffff7ffe150, image = 0x7ffff7de4fdf <_dl_lookup_symbol_x+287>}, {image_info = 0x0, image = 0x0}, {image_info = 0x7fff00000001, image = 0x0}, {image_info = 0x7fff00000001,
            image = 0x7ffff7ffe150}, {image_info = 0x100000000, image = 0x1000001c0}, {image_info = 0x0, image = 0x7ffff7de44f9 <check_match+89>}, {image_info = 0x0, image = 0x7ffff7ffe4a8}, {image_info = 0x7fffffffdf30,
            image = 0x7fffffffdf20}, {image_info = 0x24ce7d03, image = 0x4005e7}, {image_info = 0xffffffff, image = 0x7fffffffdf80}, {image_info = 0x7ffff75ed478, image = 0x7ffff7ff6b48}, {image_info = 0x770000006e, image = 0x0}, {
            image_info = 0x7fffffffdf7f, image = 0x0}, {image_info = 0x770000006e, image = 0x0}, {image_info = 0x7fff0000007c, image = 0x0}, {image_info = 0x7fffffffdfaf, image = 0x4003b0}, {image_info = 0x10000007c,
            image = 0x602040 <IdentifyImageCommand@got.plt>}, {image_info = 0x1, image = 0x1}, {image_info = 0x0, image = 0x7fffffffe608}, {image_info = 0x7fffffffe4f0, image = 0x7ffff7de98de <_dl_fixup+206>}, {image_info = 0x1,
            image = 0x0}, {image_info = 0x10a4, image = 0x7ffff75ed478}, {image_info = 0x7fffffffe3b0, image = 0x7ffff7df0f5a <_dl_runtime_resolve_xsave+138>}, {image_info = 0x400960 <IdentifyImageCommand@plt>, image = 0x7fffffffe4f0},
          {image_info = 0x7fffffffe608, image = 0x2}, {image_info = 0x60a930, image = 0x609190}, {image_info = 0x60a570, image = 0x7ffff4a171f6 <_nl_load_locale_from_archive+678>}, {image_info = 0x7fffffffef28, image = 0x7fffffffe190},
          {image_info = 0x7fffffffe2b0, image = 0xffff00001fa0}, {image_info = 0x7fffecc2e038, image = 0xf7917730}, {image_info = 0x7fffffffe2b0, image = 0x7fffecc2e000}, {image_info = 0x666, image = 0x603ea0}, {
            image_info = 0x7fffecc542e0, image = 0x3e964}, {image_info = 0x7fffecdc26d0, image = 0x36}, {image_info = 0x7fffecdc2710, image = 0x996}, {image_info = 0x7fffecc92c50, image = 0x12fa76}, {image_info = 0x7fffecdc30b0,
            image = 0x11e}, {image_info = 0x0, image = 0xffff0000ff000000}, {image_info = 0x54552e53555f6e65, image = 0x7fff00382d46}, {image_info = 0x2e2e00544d470043, image = 0x5249445a54002f}, {image_info = 0x0, image = 0x0}, {
            image_info = 0x4040404040404040, image = 0x4040404040404040}, {image_info = 0x5a5a5a5a5a5a5a5a, image = 0x5a5a5a5a5a5a5a5a}, {image_info = 0x2020202020202020, image = 0x2020202020202020}, {image_info = 0xffffffffffffffff,
            image = 0xffff00ffffffff00}, {image_info = 0x0, image = 0x0}, {image_info = 0x0, image = 0x0}, {image_info = 0x0, image = 0x0}, {image_info = 0xffff000000ff00ff, image = 0xff000000ffff00}, {image_info = 0x0, image = 0x0}, {
            image_info = 0x0, image = 0x0}, {image_info = 0x0, image = 0x0}, {image_info = 0x0, image = 0x0}, {image_info = 0x609e20, image = 0xb}, {image_info = 0xc, image = 0x7ffff4a6a10c <__GI___libc_malloc+76>}, {image_info = 0x0,
            image = 0xd}, {image_info = 0x10f, image = 0x7ffff4a15ae0 <new_composite_name+416>}, {image_info = 0x7ffff4b51c40 <_nl_category_names>, image = 0x100608fd0}, {image_info = 0x7fffffffe320,
            image = 0x7ffff4a22102 <__GI_getenv+194>}, {image_info = 0x2, image = 0x0}, {image_info = 0x0, image = 0x0} <repeats 18 times>, {image_info = 0x60a930, image = 0x0}}
        fire = <optimized out>
        pend = MagickFalse
        respect_parenthesis = MagickFalse
        status = 1
        i = 1
        count = <optimized out>
        j = 1
        k = 0
        __PRETTY_FUNCTION__ = "IdentifyImageCommand"
        __func__ = "IdentifyImageCommand"
#8  0x00007ffff769e203 in MagickCommandGenesis (image_info=image_info@entry=0x60a930, command=0x400960 <IdentifyImageCommand@plt>, argc=argc@entry=2, argv=argv@entry=0x7fffffffe608, metadata=metadata@entry=0x7fffffffe4f0,
    exception=exception@entry=0x609190) at wand/mogrify.c:172
        option = <optimized out>
        duration = <optimized out>
        serial = <optimized out>
        concurrent = <optimized out>
        regard_warnings = <optimized out>
        status = MagickTrue
        i = <optimized out>
        iterations = <optimized out>
        number_threads = <optimized out>
        n = <optimized out>
#9  0x0000000000400a2f in IdentifyMain (argv=0x7fffffffe608, argc=2) at utilities/identify.c:80
        metadata = 0x0
        exception = 0x609190
        image_info = 0x60a930
        status = <optimized out>
#10 main (argc=2, argv=0x7fffffffe608) at utilities/identify.c:93
```
</details>
&nbsp;

Also tested with the last commit (https://github.com/ImageMagick/ImageMagick/commit/276c08e0ea8027b10e823a2fadc4e414bf6e7857) but the segfault persists.

**Comment by urban-warrior** [comment]
We verified the problem is fixed in ImageMagick-6.9.9-35 Beta.  A release is imminent, perhaps by the end of the week.

**Comment by kleisauke** [comment]
Just tested ImageMagick-6.9.9-35 Beta and I confirm that this segmentation fault has been resolved. Thanks for fixing this! Looking forward to the next release.
